src/databricks/labs/remorph/reconcile/execute.py

```diff
@@ -50,21 +54,39 @@ def recon(ws: WorkspaceClient, spark: SparkSession, table_recon: TableRecon, sou
     # initialise the Reconciliation
     reconciler = Reconciliation(source, target, database_config, report_type, schema_comparator, source_dialect)
 
+    # initialise the recon capture class
+    recon_capture = ReconCapture(
+        database_config=database_config,
+        recon_id=recon_id,
+        report_type=report_type,
+        source_dialect=source_dialect,
+        ws=ws,
+        spark=spark,
+    )
+
     for table_conf in table_recon.tables:
+        recon_process_duration = ReconcileProcessDuration(start_ts=str(datetime.datetime.now()), end_ts=None)
         src_schema = source.get_schema(
             catalog=database_config.source_catalog, schema=database_config.source_schema, table=table_conf.source_name
         )
         tgt_schema = target.get_schema(
             catalog=database_config.target_catalog, schema=database_config.target_schema, table=table_conf.source_name
         )
 
-        if report_type in {"data", "row"}:
-            reconciler.reconcile_data(table_conf=table_conf, src_schema=src_schema, tgt_schema=tgt_schema)
-        elif report_type == "schema":
-            reconciler.reconcile_schema(table_conf=table_conf, src_schema=src_schema, tgt_schema=tgt_schema)
-        elif report_type == "all":
-            reconciler.reconcile_data(table_conf=table_conf, src_schema=src_schema, tgt_schema=tgt_schema)
-            reconciler.reconcile_schema(table_conf=table_conf, src_schema=src_schema, tgt_schema=tgt_schema)
+        reconcile_output = ReconcileOutput()
```

Comment:
```suggestion
        reconcile_output = None
        schema_output = None
```

Reply:
We should not assign None Sundar. Why because In the recon_capture we don't worry what report_type is we take the both reconcile_output and schema_output and store it in appropriate tables. To handle that we can initialize empty dataclass with default values. 

this should followed for exception handling also. Default dataclass to be initialized along with exception.